### PR TITLE
update openshift-sdn dep for service isolation support

### PR DESCRIPTION
**Pull request by danwinship**
bumps the openshift-sdn dep to current master, adding multitenant service isolation support, and also picking up the minion->node renaming

@rajatchopra

### CODE REVIEW

**Comment by liggitt** [comment]
split into two commits, one for the godep bump and one for the origin/pkg changes

File: pkg/cmd/server/bootstrappolicy/policy.go [hunk 1]
```diff
@@ -387,6 +387,11 @@ func GetBootstrapClusterRoles() []authorizationapi.ClusterRole {
 					Resources: util.NewStringSet("services"),
 				},
 				{
+					// Needed to get namespace<->id mappings for service isolation rules
```
Comment (liggitt) [inline pkg/cmd/server/bootstrappolicy/policy.go]: has compatibility implications, need to update release notes to note new required permission

Comment (liggitt) [inline pkg/cmd/server/bootstrappolicy/policy.go]: if this is for SDN, shouldn't it be in the "system:sdn-reader" role? nodes are already included in the "system:sdn-reader" role, so they already have permission to get/list/watch namespaces

File: Godeps/_workspace/src/github.com/openshift/openshift-sdn/ovssubnet/api/types.go [hunk 2]
```diff
@@ -31,21 +31,26 @@ type SubnetRegistry interface {
...
+	GetServices() (*[]Service, error)
+	WatchServices(receiver chan *ServiceEvent, stop chan bool) error
 }
 
 type SubnetEvent struct {
-	Type   EventType
-	Minion string
-	Sub    Subnet
+	Type EventType
+	Node string
```
Comment (liggitt) [inline Godeps/_workspace/src/github.com/openshift/openshift-sdn/ovssubnet/api/types.go]: does this rename affect the serialization/deserialization? if so, how do we do a rolling update where some nodes are old and some are new?

File: Godeps/_workspace/src/github.com/openshift/openshift-sdn/ovssubnet/registry/registry.go [hunk 3]
```diff
@@ -30,27 +30,42 @@ type EtcdSubnetRegistry struct {
...
-		min.Type = api.Added
+		nodeEvent.Type = api.Added
 	}
 
 	if key != "" {
-		_, min.Minion = path.Split(key)
-		return min
+		_, nodeEvent.Node = path.Split(key)
+
+		var node map[string]interface{}
```
Comment (liggitt) [inline Godeps/_workspace/src/github.com/openshift/openshift-sdn/ovssubnet/registry/registry.go]: where is this serialized content coming from? this method of deserializing is different than the rest of the product... I'm concerned that we'll have compatibility issues between releases with field name changes

File: Godeps/_workspace/src/github.com/openshift/openshift-sdn/ovssubnet/registry/registry.go [hunk 4]
```diff
@@ -230,6 +253,11 @@ func (sub *EtcdSubnetRegistry) GetContainerNetwork() (string, error) {
 	return resp.Node.Value, err
 }
 
+func (sub *EtcdSubnetRegistry) GetServicesNetwork() (string, error) {
+	// FIXME
+	return "172.30.0.0/16", nil
```
Comment (liggitt) [inline Godeps/_workspace/src/github.com/openshift/openshift-sdn/ovssubnet/registry/registry.go]: needs to take configured subnet

**Comment by danwinship** [comment]
cc @pravisankar for answers to the minion->node questions

File: Godeps/_workspace/src/github.com/openshift/openshift-sdn/ovssubnet/api/types.go (hunk 2 above)
Comment (rajatchopra) [inline Godeps/_workspace/src/github.com/openshift/openshift-sdn/ovssubnet/api/types.go]: This is just an in-memory object. Does not get persisted anywhere.

File: Godeps/_workspace/src/github.com/openshift/openshift-sdn/ovssubnet/registry/registry.go (hunk 3 above)
Comment (rajatchopra) [inline Godeps/_workspace/src/github.com/openshift/openshift-sdn/ovssubnet/registry/registry.go]: This deserialization is for the standalone mode of openshift-sdn (not shipped/supported for enterprise).
It does not get used in the openshift product. The actual registry used in openshift is in <origin>/plugins/osdn/

**Comment by danwinship** [comment]
Actually it looks like Ravi already submitted another PR (#4056) covering the minion->node renaming part, so I'll rebase this on top of that once that lands

**Comment by pravisankar** [comment]
[test]

**Comment by danwinship** [comment]
rebased and repushed with comments addressed.

Actually, I missed fixing GetServicesNetwork() in the Godeps/ openshift-sdn code, but that code doesn't get used anyway. The code in plugins/osdn/osdn.go has been updated to now get the service network from the master.

For the release notes update, what exactly does it need to say? Just that a new permission has been added to system:sdn-reader?

**Comment by pravisankar** [comment]
[merge]

**Comment by rajatchopra** [comment]
false failure. retry [merge]

**Comment by danwinship** [comment]
Hm? worksforme:

    danw@w541:origin (master)> ./_output/local/go/bin/openshift start master --portal-net=hey-are-you-parsing-this-argument?
    Invalid MasterConfig 
      kubernetesMasterConfig.servicesSubnet: invalid value 'hey-are-you-parsing-this-argument?', Details: must be a valid CIDR notation IP range (e.g. 172.30.0.0/16)

Oh, are you talking about the change to rel-eng/completions/bash/openshift? Hm... it looks like that's a bug in update-generated-completions.sh

**Comment by danwinship** [comment]
oh, no, I see. --portal-net is still accepted for "start master", but not for all-in-one "start" (and only the former case is tested by command_test.go)

**Pull request merged**